### DISCUSSION TITLE
IndexError: list index out of range

### DISCUSSION THREAD

**Discussion by deleep225** [deepmodeling/dpgen#602] [General]
Hi im trying to run dpgen on local machine for **init** process, Im getting the following error.

rhome/dhileept/.local/lib/python3.9/site-packages/dpgen/data/gen.py:612: UserWarning: the dpdispatcher will be updated to new version.And the interface may be changed. Please check the documents for more details
  warnings.warn(f"the dpdispatcher will be updated to new version."
Traceback (most recent call last):
  File "/rhome/dhileept/.local/bin/dpgen", line 8, in <module>
    sys.exit(main())
  File "/rhome/dhileept/.local/lib/python3.9/site-packages/dpgen/main.py", line 175, in main
    args.func(args)
  File "/rhome/dhileept/.local/lib/python3.9/site-packages/dpgen/data/gen.py", line 783, in gen_init_bulk
    make_scale(jdata)
  File "/rhome/dhileept/.local/lib/python3.9/site-packages/dpgen/data/gen.py", line 380, in make_scale
    poscar_scale(pos_src, 'POSCAR', jj)
  File "/rhome/dhileept/.local/lib/python3.9/site-packages/dpgen/data/gen.py", line 176, in poscar_scale
    if 'D' == lines[7][0] or 'd' == lines[7][0]: 
IndexError: list index out of range

my machine.json file is 
{
    "_comment":         "training on localhost ",
    "_comment" : "This is for DeePMD-kit 1.*",

    "_comment":         "fp on localhost ",
    "fp_command":       "sbatch /rhome/dhileept/script",
    "fp_group_size":    2,
    "fp_machine":       {
        "batch":        "shell",
        "work_path" :   "/rhome/dhileept/tut_dpgen/dpgen/examples/init/tmp",
        "_comment" :    "that's all"
    },
    "fp_resources":     {
        "with_mpi":     false,
        "_comment":     "that's all"
    },

    "_comment":         " that's all "

and param.json is
{
    "stages":           [1,2,3,4],
    "cell_type":        "diamond",
    "super_cell":       [1, 1, 1],
    "elements":         ["H","C"],
    "from_poscar":      true,
    "from_poscar_path": "./CH4.POSCAR",
    "potcars":          ["./POT_H", "./POT_C"],
    "relax_incar":      "./INCAR_methane.rlx",
    "md_incar" :        "./INCAR_methane.md",
    "scale":            [1.00],
    "skip_relax":       false,
    "pert_numb":        30,
    "md_nstep" :        10,
    "pert_box":         0.03,
    "pert_atom":        0.01,
    "deepgen_templ":    "../generator/template/",
    "coll_ndata":       5000,
    "_comment":         "that's all"
}

**Comment by PKUfjh** [comment]
hey, I have the same issue. Do you know how to handle this right now?

**Comment by lujunfeng1** [comment]
excuse me, I have the same error too. Do you know how to handle this right now?

**Reply by PKUfjh** [reply]
> excuse me, I have the same error too. Do you know how to handle this right now?

The issue leading to this error can vary. In my previous case, the error is due to the failure of my Vasp DFT calculation, the CONTCAR is empty. So one possible reason that you can check is your first principle calculation result.
